src/backend/distributed/executor/executor.c

```diff
@@ -388,8 +389,19 @@ CitusExecScan(CustomScanState *node)
 		bool interTransactions = false;
 		int targetPoolSize = DEFAULT_POOL_SIZE;
 
-		/* we are taking locks on partitions of partitioned tables */
-		LockPartitionsInRelationList(distributedPlan->relationIdList, AccessShareLock);
+		/*
+		 * PostgreSQL takes locks on all partitions in the executor. It's not entirely
+		 * clear why this is necessary (instead of locking the parent during DDL), but
+		 * We do the same for consistency.
+		 */
+		if (distributedPlan->operation == CMD_SELECT)
```

Comment:
@onderkalaci This is kind of a recurring theme. I wonder whether we should have a plan type e.g. `PLAN_READ` `PLAN_WRITE` `PLAN_SCHEMA_CHANGE` `PLAN_VACUUM`.

(here we can only have PLAN_READ and PLAN_WRITE)